apollo-router/src/spec/query/change.rs

```diff
@@ -1002,6 +1033,46 @@ mod tests {
 
         let query = "{ __schema { types { name } } }";
 
-        assert_ne!(hash(schema1, query), hash(schema2, query));
+        assert!(hash(schema1, query).doesnt_match(&hash(schema2, query)));
+    }
+
+    #[test]
+    fn fields_with_different_arguments_have_different_hashes() {
+        let schema: &str = r#"
+        schema {
+          query: Query
+        }
+    
+        type Query {
+          test(arg: Int): String
+        }
+        "#;
+
+        let query_one = "query { a: test(arg: 1) b: test(arg: 2) }";
+        let query_two = "query { a: test(arg: 1) b: test(arg: 3) }";
+
+        // this test won't pass until we land a fix for it
+        // assert!(hash(schema, query_one).doesnt_match(&hash(schema, query_two)));
```

Comment:
Don't forget to put this back in 🕺

Reply:
Thanks — I clarified the comment above (which confused me too)